src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  let resolved = false;
+  const handleClick = (e) => {
+    if (e.button === 0) {
+      resolved = true;
+      resolve('First promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', handleClick);
+
+  setTimeout(() => {
+    if (!resolved) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  const handleClick = (e) => {
+    e.preventDefault();
+
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', handleClick);
+  document.addEventListener('contextmenu', handleClick);
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  const handleClick = (e) => {
+    e.preventDefault();
```

Comment:
Similar to the second promise, `e.preventDefault();` is used here. Ensure this is necessary for the intended functionality, as it prevents default actions for all clicks.